feign/src/main/java/com/baeldung/feign/header/interceptor/AuthRequestInterceptor.java

```diff
@@ -0,0 +1,21 @@
+package com.baeldung.feign.header.interceptor;
+
+import java.util.UUID;
+
+import feign.RequestInterceptor;
+import feign.RequestTemplate;
+
+
+public class AuthRequestInterceptor implements RequestInterceptor {
+   
+    @Override
+    public void apply(RequestTemplate template) {
+        template.header("Authorisation", "Bearer " + this.getAuthorisationToken());
```

Comment:
I think the use of a getter for the auth token is misleading here. This approach would never work in practice. In practice, we'd construct the interceptor by either passing it the auth token to use, or by giving it an object that can fetch the current auth token.

The simplest approach is to make the auth token a String field of this class and to pass that token in on construction. That's real world, and a good example for the reader to follow.

Reply:
I thought not to make the code for AuthInterceptor too complicated with additional classes, so put one simple method in there. 
Now, I have passing an Interface to this Auth Interceptor and in runtime , i will pass one object which has simple Auth token implementation. So in nutshell, the AuthInterceptor is not generating the token rather delegating to another object instance. Definitely in actual scenario the Overriden class needs to deal with that logic in different way.